app/main.py

```diff
@@ -1,3 +1,47 @@
+from __future__ import annotations
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __float__(self) -> float:
+        return float(self.km)
+
+    def __radd__(self, rho: Any) -> Distance:
+        return self.__add__(rho)
+
+    def __add__(self, rho: Any) -> Distance:
+        return Distance(self.km + float(rho))
+
+    def __iadd__(self, rho: Any) -> Distance:
+        self.km += float(rho)
+        return self
```

Comment:
are u sure about Any for type annotation ? could we get float f.e. from dict ?) 
also what is "rho" ?) fix this everywhere, please

Reply:
rho stands for 'right-hand object", it's a common way of parameter naming when overriding a binary operator ( [left-hand object] [operator] [right-hand object] ). So, when it takes making a method (C# or python for example) to override an operator, you have 'this' ('self'/'me') and 'rho', where 'this' contains the left-hand object and 'rho' - right-hand object. For languages like C++,  when you have to use a friendly function to override most operators, you have both lho (for left-hand object) and rho parameters

Reply:
Will fix annotations in the next commit